lib/rspec/matchers/built_in/contain_exactly.rb

```diff
@@ -60,7 +60,9 @@ def convert_actual_to_an_array
         end
 
         def safe_sort(array)
-          array.sort rescue array
+          array.sort
+        rescue Exception
```

Comment:
Because RSpec::Mocks::ExpectationNotMetError subclasses Exception, not StandardError.